### PR TITLE
Create EthRequestEventSpec and unify OCRJobRuns => PipelineRuns

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

**Comment by j16r** [comment]
Mind using something a bit more descriptive for git comments than `wip` ?

**Comment by samsondav** [comment]
@j16r yup my apologies. It wasn't ready, I should have put a "not ready for review" label on this. Commit history is cleaned up now.

File: core/web/jobs_controller.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+// Index lists all jobs
+// Example:
+// "GET <application>/jobs"
+func (jc *JobsController) Index(c *gin.Context) {
+	jobs, err := jc.App.GetStore().ORM.JobsV2()
+	if err != nil {
+		jsonAPIError(c, http.StatusInternalServerError, err)
+		return
+	}
+	fmt.Println("Balls", jobs)
```
Comment (DeividasK) [inline core/web/jobs_controller.go:33]: Good ol' balls 😆 haven't seen that in a while

Comment (DeividasK) [reply core/web/jobs_controller.go:33]: We need to get Peter to Chainlink... 🤔

File: core/web/jobs_controller.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+	jsonAPIResponse(c, jobs, "jobs")
+}
+
+// Show returns the details of a job
+// Example:
+// "GET <application>/jobs/:ID"
+func (jc *JobsController) Show(c *gin.Context) {
```
Comment (DeividasK) [inline core/web/jobs_controller.go:41]: I think show doesn't work with ethrequestevent specs

File: core/web/jobs_controller.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	jobSpec := models.JobSpecV2{}
+	err := jobSpec.SetID(c.Param("ID"))
+	if err != nil {
+		jsonAPIError(c, http.StatusUnprocessableEntity, err)
+		return
+	}
+
+	jobSpec, err = jc.App.GetStore().ORM.FindOffChainReportingJob(jobSpec.ID)
```
Comment (DeividasK) [inline core/web/jobs_controller.go:49]: I think show doesn't work with ethrequestevent specs or above just needs a rename

File: core/web/jobs_controller_test.go [hunk 4]
```diff
@@ -110,11 +110,36 @@ func TestOCRJobSpecsController_Create_HappyPath(t *testing.T) {
...
+	assert.NoError(t, err)
+
+	assert.Equal(t, "example eth request event spec", job.Name.ValueOrZero())
+	assert.NotNil(t, jobSpec.PipelineSpec.DotDagSource)
+
+	// Sanity check to make sure it inserted correctly
+	require.Equal(t, models.EIP55Address("0x613a38AC1659769640aaE063C651F48E0250454C"), job.EthRequestEventSpec.ContractAddress)
+}
+
+func TestJobsController_Index_HappyPath(t *testing.T) {
```
Comment (DeividasK) [inline core/web/jobs_controller_test.go:138]: Could you add an "Eth..." job spec to this test?

File: core/web/jobs_controller_test.go [hunk 5]
```diff
@@ -127,11 +152,11 @@ func TestOCRJobSpecsController_Index_HappyPath(t *testing.T) {
 	runOCRJobSpecAssertions(t, ocrJobSpecFromFile, ocrJobSpecs[0])
 }
 
-func TestOCRJobSpecsController_Show_HappyPath(t *testing.T) {
+func TestJobsController_Show_HappyPath(t *testing.T) {
```
Comment (DeividasK) [inline core/web/jobs_controller_test.go:155]: Need a test for "Eth..." job spec

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -15,6 +15,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Bring `/runs` tab back to the operator UI
 
+### Changed
+
+- Commands for creating/managing legacy jobs and OCR jobs have changed, to accomodate additional types of jobs using the new pipeline.
+
+#### Legacy (V1) jobs
```
Comment (DeividasK) [inline docs/CHANGELOG.md:22]: Is it really legacy? Or just v1?

File: core/services/eth_request_events.go [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"gopkg.in/guregu/null.v4"
+)
+
+// EthRequestEvent is a wrapper for `models.EthRequestEvent`, the DB
+// representation of the job spec. It fulfills the job.Spec interface
+// and has facilities for unmarshaling the pipeline DAG from the job spec text.
+type EthRequestEventSpec struct {
+	// TODO: Abstract these fields?
```
Comment (DeividasK) [inline core/services/eth_request_events.go:16]: Ticket or no TODO 🙂 I also don't particularly like this given we have job spec v2 and a lot of (all?) of these fields are duplicated here, but we already had this conversation 🤷‍♂️

Comment (samsondav) [reply core/services/eth_request_events.go:16]: Well, I suppose this is a request to any reviewers to weigh in with their thoughts

**Review by se3000 — APPROVED** [review]
I like the IDEmbed. Love that this migration will help us trim "JobSpec" name to "Job", although I'm still a bit confused about some of the distinctions with the to/fromDBRow methods. Seems consistent, just found it a little confusing that it is a DBRow but accepts/returns a JobSpec.

**Comment by samsondav** [comment]
@se3000 I am also still confused about the ToDBRow/FromDBRow abstraction. I was going through a "paint by numbers" approach which seemed to work out but I find it still confusing.

I remember having a long back-and-forth with @spooktheducks about this. It's possible there is a better/cleaner way but right now I am not sure how that looks.

**Review by DeividasK — APPROVED** [review]
Nice one! 👍

**Pull request merged**